### PR TITLE
MM-31882 - Add CodeQL check

### PR DESCRIPTION
**Pull request by srkgupta**
#### Summary
Added step to run automated CodeQL checks. Once this file is merged, the codeql checks will automatically be performed on new pull requests to master branch. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31882

#### Release Note
None

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '17 2 * * 0'
```
Comment (agnivade) [inline .github/workflows/codeql-analysis.yml:10]: What is the significance of 2:17 on a Sunday?

Comment (srkgupta) [reply .github/workflows/codeql-analysis.yml:10]: That's just a random weekly time picked

Comment (agnivade) [reply .github/workflows/codeql-analysis.yml:10]: Then perhaps 0:00 is as good a time as any? 2:17 just reads odd.

Comment (srkgupta) [reply .github/workflows/codeql-analysis.yml]: Thanks. Updated the cron time.

**Review by agnivade — APPROVED** [review]

**Review by jupenur — APPROVED** [review]

**Review by sbishel — APPROVED** [review]

**Pull request merged**